server/qrops.go

```diff
@@ -0,0 +1,273 @@
+}
+
+func getQrURL(params url.Values, logger *zap.Logger) string {
+	qrURL, ok := params["url"]
+
+	if !ok || len(qrURL) == 0 {
+		logger.Error("[qrops-error] no qr url provided")
+	}
+
+	decodedURL, err := base64.StdEncoding.DecodeString(qrURL[0])
+
+	if err != nil {
+		logger.Error("[qrops-error] unable to decoding string from base64", zap.Error(err))
+	}
+
+	return string(decodedURL)
+}
+
+func getAllowProfileImage(params url.Values, logger *zap.Logger) bool {
+
+	allowProfileImage, ok := params["allowProfileImage"]
+
+	if !ok || len(allowProfileImage) == 0 {
+		// we default to false when this flag was not provided
+		// so someone does not want to allowProfileImage on their QR Image
+		// fine then :)
+		return false
+	}
+
+	LogoOnImage, err := strconv.ParseBool(allowProfileImage[0])
+
+	if err != nil {
+		// maybe for fun someone tries to send non-boolean values to this flag
+		// we also default to false in that case
+		return false
+	}
+
+	// if we reach here its most probably true, but we pass the flag back
+	return LogoOnImage
+
+}
+
+func getLevel(params url.Values, logger *zap.Logger) string {
+	level, ok := params["level"]
+
+	if !ok || len(level) == 0 {
+		// we default to MediumLevel of error correction when the level flag
+		// is not passed.
+		return "2"
+	}
+
+	return level[0]
+}
+
+func getKeyUID(params url.Values, allowProfileImages bool, logger *zap.Logger) string {
+	keyUID, ok := params["keyUid"]
+
+	// if allowProfileImage was also not passed
+	// in that case we don't care and can return an empty string back
+	if !allowProfileImages {
+		return ""
+	}
+
+	// the keyUID was not passed, which is a requirement to get the multiaccount image,
+	// so we log this error
+	if !ok || len(keyUID) == 0 {
+		logger.Error("[qrops-error] A keyUID is required to put logo on image and it was not passed in the parameters")
+	}
+
+	return keyUID[0]
+}
+
+func getImageName(params url.Values, allowProfileImages bool, logger *zap.Logger) string {
+	imageName, ok := params["imageName"]
+
+	// if allowProfileImage was also not passed
+	// in that case we don't care and can return an empty string back
+	if !allowProfileImages {
+		return ""
+	}
+
+	//if the imageName was not passed, we default to "thumbnail"
+	if !ok || len(imageName) == 0 {
+		return "thumbnail"
+	}
+
+	return imageName[0]
+}
+
+func getSize(params url.Values, logger *zap.Logger) int {
+	size, ok := params["size"]
+
+	//if the size was not passed, we return nil;
+	if !ok || len(size) == 0 {
+		return 0
+	}
+	imageToBeResized, err := strconv.Atoi(size[0])
+
+	if err != nil {
+		logger.Error("[qrops-error] could not convert string to int for size param ", zap.Error(err))
+		return 0
+	}
+
+	return imageToBeResized
+}
+
+func generateQRBytes(params url.Values, logger *zap.Logger, multiaccountsDB *multiaccounts.Database) []byte {
+
+	qrURLBase64Decoded := getQrURL(params, logger)
+	superImposeLogoOnQR := getAllowProfileImage(params, logger)
+	level := getLevel(params, logger)
```

Comment:
Should this function return a `qrcode. ErrorCorrection` type? rather than parsing the output externally, and the string variable `level` isn't used after parsing.

Reply:
hmm I was facing a stupid type error when I had tried doing that, I will try again 👍🏻

Reply:
fixed here : https://github.com/status-im/status-go/pull/3172/commits/e802e70586126844a5c6bbe242368217f222d8f0